chia/wallet/wallet.py

```diff
@@ -276,16 +276,70 @@ async def select_coins(self, amount, exclude: List[Coin] = None) -> Set[Coin]:
         unconfirmed_removals: Dict[bytes32, Coin] = await self.wallet_state_manager.unconfirmed_removals_for_wallet(
             self.id()
         )
+        # check for exact 1 to 1 coin match.
```

Comment:
This should be decomposed into another file and method, for easier code reading in the future.
For example, `coin_selection.py`, `find_exact_coin(coins .....`